FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeProcessingJob.cs

```diff
@@ -181,6 +182,12 @@ public async Task<string> ExecuteAsync(IProgress<string> progress, CancellationT
 
                 throw new RetriableJobException("Unhandled error occurred in data processing job.", ex);
             }
+            finally
+            {
+                // Compact large objects allocated in processing.
+                GCSettings.LargeObjectHeapCompactionMode = GCLargeObjectHeapCompactionMode.CompactOnce;
```

Comment:
Just curious: do we need to set the mode every time? is it possible to set it once?

Reply:
No. But set it here can make it more obvious that we will collect LOH in GC

Reply:
Seems it is necessary set it every time.

Ref:
"If you assign the property a value of [GCLargeObjectHeapCompactionMode.CompactOnce](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.gclargeobjectheapcompactionmode?view=net-7.0#system-runtime-gclargeobjectheapcompactionmode-compactonce), the LOH is compacted during the next full blocking garbage collection, and the property value is reset to [GCLargeObjectHeapCompactionMode.Default](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.gclargeobjectheapcompactionmode?view=net-7.0#system-runtime-gclargeobjectheapcompactionmode-default)."

(https://learn.microsoft.com/en-us/dotnet/api/system.runtime.gcsettings.largeobjectheapcompactionmode?view=net-7.0)